### PR TITLE
fix: wait for nav to be ready to follow push notifications

### PR DESCRIPTION
**Pull request by brainbicycle**
This PR resolves [] 

### Description

Noticed while testing some other push changes that if we try to navigate from a push before navigation is ready 
the app can crash or get an unable to load error. 

This does 2 things:
1. Check if nav is ready in Android push handling, if not save as pending notification and handle same way we do notifications to logged out users 
2. Safe unwrap all places we access nav references, bit surprised by this but it looks like in some cases `navRef.current?.getRoutes().routes` will call getRoutes and return undefined, so we need to check the value of the returned routes before access like is not safe, need to do `navRef.current?.getRoutes()?.routes`, some errors in sentry [here](https://artsynet.sentry.io/issues/?environment=production&environment=bootstrap&environment=development&environment=staging&project=5867225&query=Cannot+read+property+%27routes%27+of+undefined+os.name%3AAndroid&referrer=issue-list&statsPeriod=90d) around this

### PR Checklist

- [x] I have tested my changes on **iOS** and **Android**.
- [x] I hid my changes behind a **[feature flag]**, or they don't need one.
- [x] I have included **screenshots** or **videos**, or I have not changed the UI.
- [x] I have added **tests**, or my changes don't require any.
- [x] I added an **[app state migration]**, or my changes do not require one.
- [x] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [x] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

- fix push notifications not being followed or causing crashes - brian 

#### Dev changes

-

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
### This PR contains the following changes:

- Android user-facing changes (fix push notifications not being followed or causing crashes - brian)
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 0af8c840e6fb7cf25f52caf1994e2a0c6dc4241a
</p>

File: src/app/utils/PushNotification.ts [hunk 1]
```diff
@@ -188,7 +190,7 @@ export const handleReceivedNotification = (
         message: notification?.message?.toString(),
       })
     }
-    if (!isLoggedIn) {
+    if (!isLoggedIn || !isNavigationReady) {
```
Comment (brainbicycle) [inline src/app/utils/PushNotification.ts:193]: main change: check in push handling on android if nav is ready, if not handle as pending notification

File: src/app/utils/useInitialNotification.ts [hunk 2]
```diff
@@ -29,5 +35,5 @@ export function useInitialNotification() {
       return
     }
     handlePendingNotification(pendingNotification)
-  }, [isLoggedIn])
+  }, [isLoggedIn, isNavigationReady])
```
Comment (brainbicycle) [inline src/app/utils/useInitialNotification.ts:38]: main change: update the pending notification hook to watch nav readiness

**Review by MrSltun — APPROVED** [review]
Thank you Brian! 🙌

**Pull request merged**